### ISSUE TITLE
[P1][Sourcing][Accessibility][Multi-input][input needed] The role “input field” is not correct, because no free text input is possible. Since it is not a corresponding role in ARIA or HTML, the role should be left, but an input hint should be used. If appropriate, aria-roledescription can be used to describe the role.

### ISSUE THREAD

**Issue by I543348** [SAP/fundamental-ngx#10287]
#### Is this a bug, enhancement, or feature request?
Bug

#### Briefly describe your proposal.
The role “input field” for multi-input is not correct, because no free text input is possible. Since it is not a corresponding role in ARIA or HTML, the role should be left, but an input hint should be used. If appropriate, aria-roledescription can be used to describe the role.

#### If this is a bug, please provide steps for reproducing it.
Tested on multi-input mobile mode example

#### Is there anything else we should know?
Reported by customer and below are some references -
Conclusion: The element should not be used without adjustments, because it has many problems. Better works for multiple selection e.g. [https://www.w3.org/WAI/ARIA/apg/patterns/listbox/examples/listbox-rearrangeable/](https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.w3.org%2FWAI%2FARIA%2Fapg%2Fpatterns%2Flistbox%2Fexamples%2Flistbox-rearrangeable%2F&data=05%7C01%7Canusha.gachinamath%40sap.com%7C4df308d035254bd0feba08db936136b3%7C42f7676cf455423c82f6dc2d99791af7%7C0%7C0%7C638265817978201536%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=LyeYb%2FCReQaK%2F9OtuccSR%2BCZxh7eDujoQx92cHVFryY%3D&reserved=0) and https://sapui5.hana.ondemand.com/sdk/#/entity/sap.m.MultiComboBox/sample/sap.m.sample.MultiComboBox

**Comment by droshev** [comment]
We would need more input, scenario, links.

**Comment by nikolay-kolarov** [comment]
Hi,

This sample showcase how the control behaves on mobile device. And if this sample is taken for testing of desktop accessibility, this is wrong, as on desktop the behavior is different.
On mobile if you tap on it, a dialog will open and then the user can tap in the input and enter the data.

Could you please provide if there is any real user impact on mobile, or only the sample was tested and we are safe to close the incident?

Thanks,
Nikolay

**Comment by droshev** [comment]
@I543348 Can you follow up here?

**Comment by I543348** [comment]
I have requested ATL team to comment here as they agreed that this is an issue.
I will remind them again.
Thanks.

**Comment by droshev** [comment]
@I543348 the mobile example is not supposed to be tested productively on desktop(including Screen reader).

**Comment by I543348** [comment]
ATL Re-verification needed

**Comment by sagar7259** [comment]
Hello,

The search behavior is different from the UI5 component has here in NGX the free text does not give the search results. But this is different in UI5 multi value input.

For example: the token is 'Apple'.

NGX behavior: To search this token keyword to be used is only 'A', and if any other keyword is used then the search result will be null.
UI5 behavior: To search this token keywords can be used are 'A', 'p', 'l', 'e', hence here the free text can be entered.

If the NGX behavior holds good, then the aria-roledescription can be used to search according to the tokens present.

UI5: https://sapui5.hana.ondemand.com/#/entity/sap.m.MultiInput/sample/sap.m.sample.MultiInputMaxTokens
NGX: https://sap.github.io/fundamental-ngx/#/core/multi-input#mobile

Regards,
Sagar

**Comment by nikolay-kolarov** [comment]
Hi,

As written above, this sample is not targeted to be tested on desktop. The colleagues are mimicking how the mobile behavior is working, so that it can be easily tested.
If you do the same on mobile on this sample, at the moment when you tap in the input the dialog will be opened and there will be no such problem.

So the best think will be to change/better document this sample.

Best Regards,
Nikolay

**Comment by I543348** [comment]
Hi @nikolay-kolarov , This issue has been reported by mobile mode example.
but the use case holds good even for other multi-input samples as well, since all multi-input have input field in them.
Please correct me if i am wrong.

**Comment by nikolay-kolarov** [comment]
Hi @I543348,

Yes, but the issue is not present in the other examples. Please start typing "apple" and you will see the difference.

Best Regards,
Nikolay

**Comment by I543348** [comment]
Thanks @nikolay-kolarov , I understood your point.
However, the issue reported from customer is that the input control is not allowing user to type any random text that user may want to , but only the predefined characters are allowed and considered which matches in the list , which user will have to look at the entire list before searching.
Hence the ask is , role=input is confusing here , as user cannot type anything that he / she wants rather some description is needed to guide the user accordingly , i also discussed this issue with ATL. team (@sagar7259 ) and he agrees that this needs to be fixed as an issue.

Pls let me know if you need more clarifications.

https://github.com/SAP/fundamental-ngx/assets/88182358/ee618b44-166e-40b5-a04a-c205fc730d0f


Pls find the attached recording

**Comment by nikolay-kolarov** [comment]
Hi @I543348,

Thanks for the video. I tested it and also on the video is visible that you can enter any characters that you want from the keyboard.

Do you mean that it is not possible to turn them into tokens? E.g. you want to add a Watermelon as a token, but as it is not in the list you cannot do it in this sample?

Thanks,
Nikolay

**Comment by I543348** [comment]
Yes exactly @nikolay-kolarov , not all typed text can become a "token" hence it cannot be treated as a free text , as user cannot type random text , and some guidance in the form of aria-roledescription should be given to disabled users such as blind users that this input field , needs a text that matches the list item only.

**Comment by sagar7259** [comment]
@droshev in the example https://github.com/SAP/fundamental-ngx/issues/10055 we can add new tokens using any free text. but the same cannot be done for all other examples of the multi value inputs.

Here the issue is not related to adding tokens. It can be added using free text or as per the tokens already present within them. The main issue is that the users are not conveyed weather they can enter free text or not.

For the input fields where free texts can be entered, there we don't required any aria-role description attribute. But this attribute is very much required for the multi inputs where only its respective tokens can be searched.

@rvamudha could you please reopen this ticket.?

Regards,
Sagar

**Comment by droshev** [comment]
The component offers both options. It is up to the application to allow either of the option. If there is a requirement then the application can handle this
